src/scalar_4x64_impl.h

```diff
@@ -96,6 +96,17 @@ static int secp256k1_scalar_add(secp256k1_scalar *r, const secp256k1_scalar *a,
     return overflow;
 }
 
+static void secp256k1_scalar_numsub(secp256k1_scalar *r, const secp256k1_scalar *a, const secp256k1_scalar *b) {
+    int c0 = a->d[0] < b->d[0];
+    int c1 = a->d[1] < b->d[1] || (c0 && a->d[1] == b->d[1]);
+    int c2 = a->d[2] < b->d[2] || (c1 && a->d[2] == b->d[2]);
+
+    r->d[3] = a->d[3] - b->d[3] - c2;
+    r->d[2] = a->d[2] - b->d[2] - c1;
+    r->d[1] = a->d[1] - b->d[1] - c0;
+    r->d[0] = a->d[0] - b->d[0];
+}
+
```

Comment:
I think something more like _scalar_add using an int128 accumulator would be clearer and probably faster.